### PR TITLE
Add retry on uplaod failure.

### PR DESCRIPTION
**Pull request by xinyanw409**
Description:
When there is network issue some worker nodes in the k8s cluster, the data manager pods running on top of them will be unable to upload to S3, this change add retry on upload failure.

1.Change Upload CRD. Change UploadPhaseFailed to UploadPhaseUploadFailed and UploadPhaseCleanupFailed to distinguish different kind of failure. Cleanup Failure will not lead to retry.
Add new field RetryCnt for user tracking and debug.
2.Increase backoff on retries up to max of 1 hour between retries.

Test:
Tried to run velero backup manually multiple times to hit the upload failure, but this failure is too random to hit. So now working on gmock unit test for this change and will update the test file later. Currently post out review for code change.
CICD pass.

### CODE REVIEW

File: pkg/apis/veleroplugin/v1/upload.go [hunk 1]
```diff
@@ -75,6 +76,11 @@ type UploadStatus struct {
 	// If the DataManager couldn't process Upload for some reason it will be picked up by another
 	// node.
 	ProcessingNode string `json:"processingNode,omitempty"`
+
+	// RetryCnt records the retry count for enqueing an Upload CR when there
+	// is a newtork issue which cause an UploadFailed.
+	// +optional
+	RetryCnt int32 `json:"retryCnt,omitempty"`
```
Comment (lintongj) [inline pkg/apis/veleroplugin/v1/upload.go]: better to use RetryCount

File: pkg/controller/generic_controller.go [hunk 2]
```diff
@@ -137,7 +137,7 @@ func (c *genericController) processNextWorkItem() bool {
 	c.logger.WithError(err).WithField("key", key).Error("Error in syncHandler, re-adding item to queue")
 	// we had an error processing the item so add it back
 	// into the queue for re-processing with rate-limiting
-	c.queue.AddRateLimited(key)
+	c.queue.AddAfter(key, 1 * time.Hour)
```
Comment (lintongj) [inline pkg/controller/generic_controller.go]: please implement the exponential backoff as we discussed in the meeting.

File: pkg/controller/upload_controller.go [hunk 3]
```diff
@@ -330,11 +331,27 @@ func patchUploadFailure(c *uploadController, req *pluginv1api.Upload, msg string
 	// update status to Failed
 	log := loggerForUpload(c.logger, req)
 	var err error
-	req, err = c.patchUpload(req, func(r *pluginv1api.Upload) {
-		r.Status.Phase = pluginv1api.UploadPhaseFailed
-		r.Status.CompletionTimestamp = &metav1.Time{Time: c.clock.Now()}
-		r.Status.Message = msg
-	})
+	if strings.Contains(msg, "Failed to clean up local snapshot") {
```
Comment (lintongj) [inline pkg/controller/upload_controller.go]: It is not a good idead to use string match to identify error type. You can make two different methods for patching different failures.

File: pkg/apis/veleroplugin/v1/upload.go (hunk 1 above)
Comment (xinyanw409) [reply pkg/apis/veleroplugin/v1/upload.go]: Change to RetryCount.

**Pull request closed without merging**